Format UG and DG properly, and add use cases (#49)
AY2021S1-CS2103T-T10-4/tp: docs/DeveloperGuide.md

### Use cases

| Use Case ID | Description                      |
|-------------|----------------------------------|
| UC01        | Add a student                    |
| UC02        | List students                    |
| UC03        | Edit a student                   |
| UC04        | Find a student                   |
| UC05        | Delete a student                 |
| UC06        | Clear all students               |
| UC07        | Add a class                      |
| UC08        | List students within a class     |
| UC09        | Edit a class                     |
| UC10        | Find a class                     |
| UC11        | Delete a class                   |

> | UC12        | Clear all classes                |
> | UC13        | Link a student to a class     |
> | UC14        | Unlink a student from a class |

Reviewer: ```suggestion
| UC12        | Clear all classes                |
| UC13        | Link a student to a class.       |
| UC14        | Unlink a student from a class    |
```

> | UC13        | Linking a student to a class     |
> | UC14        | Unlinking a student from a class |

Reviewer: ```suggestion
| UC13        | Link a student to a class        |
| UC14        | Unlink a student from a class    |
```

      Use case resumes at step 2.

**Use case: UC04 - Find a student**

**MSS**

1.  User requests to find a student.
2.  User provides the parameters to find student.
3.  Tutor's Pet finds the student.

> 4.  Student is found by Tutor's Pet.

Reviewer: Technically Tutor's Pet can return more than one student found, so maybe it would be good to mention that it shows all matching students.

Reviewer: Applies to finding classes as well.

Author: Resolved by rewording expression to show that Tutor's Pet can return more than one user found

    Use case ends.

**Extensions**

* 2a. The parameters provided are invalid.

    * 2a1. Tutor's Pet shows an error message.

      Use case ends.

> * 3a. Student not found.

Reviewer: ```suggestion
* 3a. No students found.
```

>     * 3a1. Tutor's Pet shows that student is not found.

Reviewer: Would be good to mention that it is a message, otherwise might be confusing how to show an absence of something.

  Use case skips to step 5.

**Use case: UC06 - Clear all students**

**MSS**

1. User requests to clear all students.

> 2. Tutor's Pet clears all students.

Reviewer: Should we keep it consistent and use `delete` instead of `clear`?

Reviewer 2: I think `clear` is used to clear all students though... `delete` is meant to delete a specific student...

Author: Will keep clear instead of changing to delete

      Use case resumes at step 2.

**Use case: UC12 - Clear all classes**

**MSS**

1. User requests to clear all classes.
2. Tutor's Pet clears all classes.

    Use case ends.

> **Use case: UC13 - Linking a student to a class**

Reviewer: ```suggestion
**Use case: UC13 - Link a student to a class**
```

      Use case resumes at step 4.

> **Use case: UC14 - Unlinking a student from a class**

Reviewer: ```suggestion
**Use case: UC14 - Unlink a student from a class**
```
